test/csuite/wt5212_incremental_backup/main.c

```diff
@@ -0,0 +1,779 @@
+            }
+        }
+        scan_end_check(ret == WT_NOTFOUND);
+        /* Done processing this file. Close incremental cursor. */
+        testutil_check(incr_cur->close(incr_cur));
+
+        /* Close file descriptors if they're open. */
+        if (rfd != -1) {
+            testutil_check(close(rfd));
+            testutil_check(close(wfd));
+            // Reset file descriptors to default value after closing.
+            rfd = wfd = -1;
+        }
+        /*
+         * For each file, we want to copy the file into each of the later incremental directories so
+         * that they start out at the same for the next incremental round. We then check each
+         * incremental directory along the way.
+         */
+        for (j = i; j <= MAX_ITERATIONS; j++) {
+            testutil_check(__wt_snprintf(h, sizeof(h), "%s.%d", home_incr, j));
+            if (strncmp(filename, WTLOG, WTLOGLEN) == 0)
+                testutil_check(__wt_snprintf(buf, sizeof(buf), "cp %s/%s/%s %s/%s/%s", home,
+                  logpath, filename, h, logpath, filename));
+            else
+                testutil_check(
+                  __wt_snprintf(buf, sizeof(buf), "cp %s/%s %s/%s", home, filename, h, filename));
+            testutil_check(system(buf));
+        }
+    }
+    scan_end_check(ret == WT_NOTFOUND);
+
+    /* Done processing all files. Close backup cursor. */
+    testutil_check(backup_cur->close(backup_cur));
+    finalize_files(flist, count);
+    free(tmp);
+    /*! [incremental backup using block transfer]*/
+}
+
+static void
+add_data_validate_backups(WT_SESSION *session)
+{
+    int i, j;
+    printf("Adding initial data\n");
+    add_work(session, 0, 0, uri, uri2);
+
+    printf("Taking initial backup\n");
+    take_full_backup(session, 0);
+
+    testutil_check(session->checkpoint(session, NULL));
+
+    for (i = 1; i < ITERATIONS_MULTIPLIER; i++) {
+        printf("Iteration %d: adding data\n", i);
+        /* For each iteration we may add work and checkpoint multiple times. */
+        for (j = 0; j < i; j++) {
+            add_work(session, i, j, uri, uri2);
+            testutil_check(session->checkpoint(session, NULL));
+        }
+
+        /*
+         * The full backup here is only needed for testing and comparison purposes. A normal
+         * incremental backup procedure would not include this.
+         */
+        printf("Iteration %d: taking full backup\n", i);
+        take_full_backup(session, i);
+        /*
+         * Taking the incremental backup also calls truncate to archive the log files, if the copies
+         * were successful. See that function for details on that call.
+         */
+        printf("Iteration %d: taking incremental backup\n", i);
+        take_incr_backup(session, i);
+
+        printf("Iteration %d: dumping and comparing data\n", i);
+        testutil_check(compare_backups(i, uri));
+    }
+}
+
+/*
+ * Remove all the records, take backups and validate
```

Comment:
All comments should be complete sentences, ending in a period.

Reply:
Fixed